src/logic/map_objects/map_object.cc

```diff
@@ -158,10 +158,33 @@ ObjectManager::~ObjectManager() {
  * Clear all objects
  */
 void ObjectManager::cleanup(EditorGameBase& egbase) {
+	// If all wares (read: flags) of an economy are gone, but some workers remain,
+	// the economy is destroyed before workers detach. This can cause segfault.
+	// Destruction happens in correct order after this dirty quickie.
+	// Run at the end of game, algorithmic efficiency may be what it is.
+	const std::vector<MapObjectType> killusfirst{MapObjectType::WATERWAY,    MapObjectType::FERRY,
+	                                             MapObjectType::FERRY_FLEET, MapObjectType::SHIP,
+	                                             MapObjectType::SHIP_FLEET,  MapObjectType::PORTDOCK,
+	                                             MapObjectType::WORKER};
+	for (auto moi : killusfirst) {
+
+		bool go_on = true;
```

Comment:
You can remove the `go_on` variable and replace "go_on = false" with `break` below.

Also please make `killusfirst` a `static const` variable.

Reply:
Yes, sure. I'll wait for Travis to complete the first commit, then push again.

The list of things to nuke first is a bit overkill, for this particular regression test case. But then, if arbitrary order is fine, the going safe should not harm I guess.